pkg/liquidity-mining/contracts/gauges/avalanche/AvalancheRootGaugeFactory.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-interfaces/contracts/liquidity-mining/IAvalancheBridgeLimitsProvider.sol";
+import "@balancer-labs/v2-interfaces/contracts/vault/IVault.sol";
+
+import "@balancer-labs/v2-solidity-utils/contracts/helpers/SingletonAuthentication.sol";
+
+import "../BaseGaugeFactory.sol";
+import "./AvalancheRootGauge.sol";
+
+contract AvalancheRootGaugeFactory is IAvalancheBridgeLimitsProvider, BaseGaugeFactory, SingletonAuthentication {
+    uint256 private _minBridgeAmount;
+    uint256 private _maxBridgeAmount;
+
+    event AvalancheBridgeLimitsModified(uint256 minBridgeAmount, uint256 maxBridgeAmount);
+
+    constructor(
+        IVault vault,
+        IMainnetBalancerMinter minter,
+        IMultichainV4Router multichainRouter,
+        uint256 minBridgeAmount,
+        uint256 maxBridgeAmount
+    ) BaseGaugeFactory(address(new AvalancheRootGauge(minter, multichainRouter))) SingletonAuthentication(vault) {
```

Comment:
Another option to make this more easy to test would be to create the implementation outside the factory, and send it as a parameter. Then, you wouldn't need the mock factory, as you could just send in the mock gauge implementation.

This would be a difference with respect to other root gauges, so I'm not entirely convinced it's the best idea. Feel free to take it or leave it.